billing/credit/service.go

```diff
@@ -32,15 +32,21 @@ func (s Service) Add(ctx context.Context, cred Credit) error {
 		return fmt.Errorf("credit amount is negative")
 	}
 	// check if already credited
-	if t, err := s.transactionRepository.GetByID(ctx, cred.ID); err == nil && t.ID != "" {
+	t, err := s.transactionRepository.GetByID(ctx, cred.ID)
+
+	if err != nil {
```

Comment:
this will fail to add transaction if it returns not found from repository for this id. We want to do opposite of it.

Reply:
Yup, missed that. Added a check for it.